internal/transport/http2_server.go

```diff
@@ -414,6 +440,20 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 		return false
 	}
 
+	// "If :authority is missing, Host must be renamed to :authority." - A41
+	if len(mdata[":authority"]) == 0 {
+		// No-op if host isn't present, no eventual :authority header is a valid
+		// RPC.
+		if host, ok := mdata["host"]; ok {
+			mdata[":authority"] = host
+			delete(mdata, "host")
+		}
```

Comment:
This is also a behavior change.  Is this absolutely necessary?  If not, we should not do it IMO, as it's possible users were relying on their "host" header being propagated as-is.

Reply:
This is necessary. The paragraph defining all this logic in regards to :authority and host in A41 starts with this line: "The core gRPC implementation (not just xDS or RBAC)". Thus, this change needs to happen here. wdyt @ejona86?

Reply:
It is necessary to make sure that the authz code and the application are observing the same authority/host. If someone is relying on the host header being propagated as-is, then there would be multiple security concerns. It seems quite broken for code to expect host and ignore the authority; if code is looking at host, I'd expect it to look at authority as well. If code is purposefully having different host and authority, then I'd have severe security concerns for their code.

Reply:
It's definitely unfortunate that "host" has such a plain-looking name and shares a namespace with user-accessible metadata.